components/x-follow-button/stories/knobs.js

```diff
@@ -0,0 +1,48 @@
+// To ensure that component stories do not need to depend on Storybook themselves we return a
+// function that may be passed the required dependencies.
+module.exports = (data, { text, boolean, selectV2 }) => {
+	const Groups = {
+		Text: 'Text',
+		Status: 'Status',
+		Flags: 'Flags',
+		ExtraClasses: 'ExtraClasses'
+	};
+
+	const Text = {
+		buttonText () {
+			return text('Default text', data.buttonText, Groups.Text);
+		},
+		alternateText () {
+			return text('Alternate text', data.alternateText, Groups.Text);
+		},
+		name () {
+			return text('Name of what we add', data.name, Groups.Text);
+		}
+	}
+
+	const Status = {
+		isSelected () {
+			return boolean('isSelected', data.isSelected, Groups.Status);
+		}
+	}
+
+	const Flags = {
+		followPlusDigestEmail () {
+			return boolean('followPlusDigestEmail', data.followPlusDigestEmail, Groups.Flags);
+		}
+	}
+
+	const ExtraClasses = {
+		extraButtonClasses () {
+			return text('extraButtonClasses', data.extraButtonClasses, Groups.ExtraClasses);
+		},
+		extraFormClasses () {
+			return text('extraFormClasses', data.extraFormClasses, Groups.ExtraClasses);
+		},
```

Comment:
I would really really like to bin these extra classes props if possible, whenever apps have to "reach in" to a component it gets messy.

Reply:
What should we do in this case if we want to pass some additional classes to the button?

Reply:
Is that a current use case that needs to be preserved? Passing class names breaks encapsulation and should be avoided... alternatively, if there are a set of _known_ modifiers then we could allow for that (e.g. https://github.com/Financial-Times/x-dash/blob/master/components/x-teaser/stories/knobs.js#L195-L207)

Reply:
Nope, nothing known yet, but ok - will extra classes =)